### ISSUE TITLE
Wiring Arduino micro

### ISSUE THREAD

**Issue by lamarqueb** [philipellisis/arduino-virtual-pinball-board#11]
Good morning

congratulations for your work. I'm looking to make a PCB with your system. I'm not sure I understand where everything connects

for the output and accelerometer extension I put them on SDA and SCL in series as in the first diagram.
for the three 74HC165 I'm not sure where or for the plunger can you let me know thanks in advance.
![274564465-ffee137a-5e32-429d-85fc-629557c13baf](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/c9bf678b-c3e3-4a65-ab09-c8dcc1cb2fab)
![Sans titre-1](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/363cd16d-3bea-4016-a40e-d322b611c22b)
![Sans titre-2](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/cc017faf-b5e0-422f-ba5d-03f2310ef321)

**Comment by samwyze** [comment]
I'm not entirely sure, but from looking through the code...

On the 74hc165 I believe it should go:
hc165 Pin 1(Shiftload) to Arduino D4
hc165 Pin 2(Clock) to Arduino D1(TX)
hc165 Pin 9(QH) to Arduino D0(RX).
All the netport pins should have 10k resistors going to VCC as pullups and switch to ground

The plunger potentiometer wiper is on Arduino A5, I don't think you need to wire VCC to the plunger pot as its using the arduino input_pullup command, so just the wiper and gnd.

**Comment by lamarqueb** [comment]
like that. 
![Sans titre-1](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/33923472-495b-46cd-a5b6-59046421bbe3)

**Comment by samwyze** [comment]
Like this:
![output](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/943581/d4c32f6b-3278-47a8-aef4-120d5a57ca17)

**Comment by lamarqueb** [comment]
![Sans titre-1](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/16aa268b-f8b3-4c73-9a41-887df9a09861)

**Comment by lamarqueb** [comment]
i need only arduino micro or Pro micro it's ok ? but now the pin are different.
https://fr.aliexpress.com/item/1005003643595104.html?spm=a2g0o.order_list.order_list_main.41.6a945e5b47gpSv&gatewayAdapt=glo2fra

**Comment by samwyze** [comment]
You want the Arduino Micro. Not the pro, it doesn't break out all the pins you want

**Comment by lamarqueb** [comment]
yes I saw that afterwards. for example the A5 is not there for the plunger.

**Comment by lamarqueb** [comment]
![Sans titre-1](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/a666aef1-3e12-49b7-ae40-2324e0b160f3)

**Comment by lamarqueb** [comment]
![Sans titre-3](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/ac3a7d50-0512-41c5-a6cd-772e956e0a82)

**Comment by philipellisis** [comment]
Very nice!

**Comment by samwyze** [comment]
Well done, looks awesome. Much tidier than my stripboard version 😂

**Comment by lamarqueb** [comment]
> Very nice!

thank you I hope the routing is good for the arduino micro pins. I don't know much about programming. but I hope it will work without too much modification or even any of the code.

**Comment by lamarqueb** [comment]
> Very nice!
hi !
please you can tell me the good pinout for 
74HC165 to arduino
PCA 9685 to arduino (SCL SDA ?)
MPU6050 to arduino (SCL, SDA ?)

Best regards

Benoit

**Comment by lamarqueb** [comment]
Hello everyone

I received my cards and everything works perfectly 24 buttons, 16 outputs, plunger and accelerometer. I have a little problem, all the outputs are on when they are inactive and off when they are active. Can't we reverse that? I have cards that work with LVT817 octocoupler and therefore it is not compatible.
![20240424_160819](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/9e670c05-f0de-4ccc-9559-db47ce36080f)

**Comment by matgenois** [comment]
Is these outputs are outputs id 1 to 15 (so output that are directly from the arduino) ? or 16 to 32 (output from the pca9685) ?

**Comment by matgenois** [comment]
Pinone have those output
1-16 are output directly from the Arduino. Those are Off by default;
17-32 are output from the first PCA9685 and are for the "button boards" so output are on by default so buttons are Lighted;
>33 are driven by two more PCA9685 and are off by default.

**Comment by matgenois** [comment]
I used outputs 1 to 15 to drive my shaker motor and solenoid

You can by modifying output.cpp
```
    // if the output is the button board output, then invert the value
    if (outputId < 31) {
      outputValue = 255 - outputValue;
    }
```

**Comment by lamarqueb** [comment]
i replace 0 by 255 and 255 by 0 ?

**Comment by lamarqueb** [comment]
when I update the firmware with the software I return to my active outputs. If I don't update I can't save.
it's simpler if I go through the outputs 0-15 you will have the pins

**Comment by lamarqueb** [comment]
I found the 15 pins outputs directly on the arduino, it's simpler for me.

**Comment by matgenois** [comment]
You could write the output pins here for the next one trying to design his own board :)

**Comment by lamarqueb** [comment]
i'll share my schéma when all is finish and ok
![21](https://github.com/philipellisis/arduino-virtual-pinball-board/assets/9132966/5b92b7df-8710-468a-b5b4-7b9ca43af523)
 board 1.0 with 15 outputs (5 five on board and 10 on pins) accelerometer on separate card, purchased on aliexpress, (-2€) if I put it on the main card (7€).
everything works perfectly on the tests 
let's get started...

**Comment by JRX1138** [comment]
I would appreciate the diagram I'm not going anywhere near this far at least not yet. I'm using a Leonardo to control plunger, mpu6050 and about 8 buttons. I have the plunger and nudge-tilt functioning. The pinout would be enough to get me finished, I've hunted for a couple days at least before reaching out to CSD. Your board looks amazing.

**Comment by JRX1138** [comment]
lamarqueb did a great job on the Magic Pincab in working it through.

**Comment by lamarqueb** [comment]
thanks

**Comment by Shagen05** [comment]
lamarqueb this is what i was looking for.
Would you maybe share your diagram (schematics), that would be amazing.
